test/assertions/assertNoMutations.js

```diff
@@ -0,0 +1,24 @@
+const Code = require('code');
+
+const { expect}  = Code;
+
+
+/** Asserts that all the passed entities weren't mutated after executing the passed procedure
+ *
+ * @param {object[]} entities
+ * @param {Function} procedure
+ * @returns {Promise<void>}
+ */
+async function assertNoMutations (entities, procedure) {
```

Comment:
I used it kinda to make sure that a function is pure and doesn't mutate the passed parameters.
As for me, it will be overhead in wrapping the majority of our objects to Immutable entities.